File: keiner_lindarte_aro/instructor_henry/mis_programas/dart/orientado_a_objetos/8_clase_de_28_agosto/composicion.dart
class Bateria{}

class Telefono {

  Bateria? _bateria;

  Telefono(Bateria b){
    this._bateria = b;
  }

  Bateria add_bateria(){
    return this._bateria!;
  }

}

void main(){

  Bateria bateria_iphone = new Bateria();

  Telefono iphone_15 = new Telefono(bateria_iphone);
  
  print(iphone_15.add_bateria());

}

File: keiner_lindarte_aro/instructor_henry/mis_programas/ruby/orientado_a_objetos/8_clase_de_28_agosto/composicion.rb
class Bateria
end

class Telefono

  attr_accessor :bateria
  private :bateria

  def initialize(b)
    self.bateria = b
  end

  def add_bateria()
    return self.bateria
  end

end

bateria_iphone = Bateria.new()

iphone_15 = Telefono.new(bateria_iphone)

puts iphone_15.add_bateria()